src/compiler/generate-tests-helper.xsl

```diff
@@ -41,7 +42,9 @@
 <xsl:template match="*" as="element()+" mode="test:generate-variable-declarations">
   <xsl:param name="var" as="xs:string" required="yes" />
   <xsl:param name="type" as="xs:string" select="'variable'" />
-
+  <xsl:param name="pending" select="()" tunnel="yes" as="node()?"/>
+  <xsl:variable name="variable_is_pending" as="xs:boolean"
+    select="self::x:variable and not(empty($pending|ancestor::x:scenario/@pending) or exists(ancestor::*/@focus))"/>
   <xsl:variable name="var-doc" as="xs:string?"
     select="if (node() or @href) then concat($var, '-doc') else ()" />

```

Comment:
I see two warnings `SXWN9001: A variable with no following sibling instructions has no effect` in running `xspec-variable.xspec`.

Maybe
```suggestion
     select="if (not($variable_is_pending) and (node() or @href)) then concat($var, '-doc') else ()" />
```

Reply:
@AirQuick 
I see what you mean, in the compiled xspec-variable.xsl file. Should be fixed in 428a7116e8a7072f377bd9cd01dacfdb2736426d.

Reply:
Thanks. The warnings have gone away.